packages/react-vapor/src/components/modalWizard/ModalWizard.tsx

```diff
@@ -13,6 +14,7 @@ export interface ModalWizardProps
     extends Omit<IModalCompositeOwnProps, 'modalBodyChildren' | 'validateShouldNavigate'> {
     id: string;
     onFinish?: () => unknown;
+    onFinishAsync?: () => Promise<void>;
```

Comment:
:thinking: `onFinishAsync` looks like it is something executed on finish that does not require to be synchronous. It does not imply "validating" the modal to me.

Should this be named `onFinishValidation`? In fact, can this feature be supported by the `validateStep` method instead? :thinking:

Reply:
Interesting thought! I'm not sure if it can be featured by the `validationStep` method because we would need to trigger the promise to return something to let us know that the last step cannot be completed. Perhaps @gdostie could chime in on this one.

I think the name `onFinishValidation` might be a little misleading as we are not really validating something but to wait for a promise to finish. The hope is that we catch an error that might come from the back end to prevent the modal from closing prematurely and waiting til the `onFinishAsync` method is complete.

Reply:
My main issue with this is the fact that `onFinishAsync` overlaps a bit too much with the other properties and it can confuse our users.

If I end up using the component, I won't know at first glance that:

* `onFinishAsync` can throw an error to cancel the close action
* `onFinishAsync` will **not** be executed if I set `onFinish`
* `onFinish` *can't* cancel the close action like `onFinishAsync`

I really think it should be included in `onFinish`. Having a second attribute would just cause more confusion. I think we can do this in a non-breakable way, but even then, it is a pretty new component we might even be able to afford a breaking change.